src/ompparser.yy

```diff
@@ -2318,10 +2318,17 @@ mapper_identifier_optseq : type_var
 mapper_identifier : IDENTIFIER_DEFAULT { ((OpenMPDeclareMapperDirective*)current_directive)->setIdentifier(OMPD_DECLARE_MAPPER_IDENTIFIER_default);}
                   | EXPR_STRING { ((OpenMPDeclareMapperDirective*)current_directive)->setIdentifier(OMPD_DECLARE_MAPPER_IDENTIFIER_user); ((OpenMPDeclareMapperDirective*)current_directive)->setUserDefinedIdentifier($1); }
                   ;
-
-type_var : EXPR_STRING { ((OpenMPDeclareMapperDirective*)current_directive)->setTypeVar($1); }
+         
+type_var : EXPR_STRING { if (user_set_lang == Lang_Fortran || auto_lang == Lang_Fortran) { yyerror("The syntax should be \"type :: var\" in Fortran"); YYABORT; } else ((OpenMPDeclareMapperDirective*)current_directive)->cutTypeAndVar($1); }
+         | declare_mapper_type DOUBLE_COLON declare_mapper_var {if (user_set_lang == Lang_Fortran || auto_lang == Lang_Fortran) {} else {yyerror("The syntax should be \"type var\" in C"); YYABORT; } }
```

Comment:
Please revise the branching here according to Dr. Yan's comments. Also, if one branch is not needed (line 2323 `if`), revise the statement to keep only one branch.